### PR TITLE
Adding RenderAnchor for rendering

### PR DESCRIPTION
**Pull request by VirxEC**
Allows rendering relative to moving physics objects

Not supported on poly lines :(

### CODE REVIEW

**Comment by swz-git** [comment]
i think RenderAnchor is more clear

**Comment by VirxEC** [comment]
It's just too long of a name IMO which is why I personally like "Anchor", at least RAnchor is a nice balance
image
Maybe we can poll it in discord

**Comment by NicEastvillage** [comment]
Well, nothing stops the interfaces or the botmakers from implementing a nicer/less wordy API

File: rendering.fbs [hunk 1]
```diff
@@ -19,9 +19,28 @@ table Color {
   b:ubyte = 0;
 }
 
+table BallAnchor {
+  local:Vector3 (required);
+}
+
+table CarAnchor {
+  index:uint;
```
Comment (swz-git) [inline rendering.fbs:27]: why is this not required?

Comment (VirxEC) [reply rendering.fbs:27]: i just pushed a commit that messed this up - what part are you talking about?

Comment (VirxEC) [reply rendering.fbs:27]: the code you highlight just doesn't point to anything
image

Comment (VirxEC) [reply rendering.fbs:27]: unless you do mean `index`? all primitive types like `uint` are required by default (and actually can't be optional at all)

**Review by TangilJ — APPROVED** [review]

File: rendering.fbs (hunk 1 above)
Comment (swz-git) [reply rendering.fbs:27]: lgtm then

**Pull request merged**